testing-cci-datasets/generate_summary.py

```diff
@@ -68,123 +71,49 @@ def get_list_of_ecvs(data_sets):
 
 
 def count_success_fail(data_sets, ecv):
-    summary_dict = {}
-    summary_columns = ['supported', 'open(1)',
-                       'open_temp(2)', 'open_bbox(3)', 'cache(4)', 'map(5)']
-    supported = 0
-    not_supported = 0
-    open_success = 0
-    open_fail = 0
-    open_temp_success = 0
-    open_temp_fail = 0
-    open_bbox_success = 0
-    open_bbox_fail = 0
-    cache_success = 0
-    cache_fail = 0
-    visualize_success = 0
-    visualize_fail = 0
+    summary_row_new = {}
+    summary_columns = ['supported',
+                       'open(1)',
+                       'open_temp(2)',
+                       'open_bbox(3)',
+                       'cache(4)',
+                       'map(5)']
+    for i in summary_columns:
+        summary_row_new[i] = 0
+        summary_row_new[f'{i}_failed'] = 0
+        summary_row_new[f'{i}_percentage'] = 0
+
     if 'ALL_ECVS' not in ecv:
         for dataset in data_sets:
             if ecv in dataset['ECV-Name']:
-                if 'yes' in dataset['supported']:
-                    supported += 1
-                    if 'yes' in dataset['open(1)']:
-                        open_success += 1
-                    else:
-                        open_fail += 1
-
-                    if 'yes' in dataset['open_temp(2)']:
-                        open_temp_success += 1
-                    else:
-                        open_temp_fail += 1
-
-                    if 'yes' in dataset['open_bbox(3)']:
-                        open_bbox_success += 1
+                for column in summary_columns:
+                    if 'yes' in dataset[column]:
+                        summary_row_new[column] += 1
                     else:
-                        open_bbox_fail += 1
+                        column_failed = f'{column}_failed'
+                        summary_row_new[column_failed] += 1
```

Comment:
This is a code snippet that appears twice. Can we try to move it to a function?

Reply:
Agreed, included a function for that now.